### DISCUSSION TITLE
How to change baud rate of CAN in lpcxpresso55s16 zephyr board file

### DISCUSSION THREAD

**Discussion by Aravinthkalai** [zephyrproject-rtos/zephyr#59837] [Q&A] [answered]
Hello Everyone !

I'm trying to change the baud rate of CAN controller , but it's not working by default the baud rate is 125kb , if I try to change 1Mbps it's not working but if I downgrade the baud rate let say 100kb or 50kb it's working , I'm sharing the lpc55s16-eval.dtsi file's  can 125kb baud rate implementation part, if I try to change the value from 125000 to 1000000 it's not working , I searched on data sheet there is no information about the maximum baud rate of lpcxpresso55s16 , please help me to fix this issue.

zephyr version : 3.3.99
Board : lpcxpresso55s16

&can0 {
	status = "okay";
	bus-speed = <125000>;
	bus-speed-data = <125000>;
	pinctrl-0 = <&pinmux_can0>;
	pinctrl-names = "default";
	can-transceiver {
		max-bitrate= <5000000>;
	};
};

**Comment by henrikbrixandersen** [comment]
> I'm trying to change the baud rate of CAN controller , but it's not working by default the baud rate is 125kb , if I try to change 1Mbps it's not working but if I downgrade the baud rate let say 100kb or 50kb it's working

Please be more specific than just stating "it's not working". Include relevant error logs, test setup details, logic analyzer captures, ... that may assist people in understanding your issue better.

**Comment by henrikbrixandersen** [comment]
> I searched on data sheet there is no information about the maximum baud rate of lpcxpresso55s16

The LPC55xxx series use a Bosch M_CAN based IP for their CAN controller. It supports an arbitration/classic CAN bitrate up to 1Mbit/s and a data phase CAN bitrate of up to 8Mbit/s.

**Comment by Aravinthkalai** [comment, marked as answer]
Hello Everyone !
Actually the problem is with hardware by default lpcxpresso55s16 board doesn't have terminal resistors closed by default , we need to close the jumper JP23 and JP24  , It's working now!
